app/domain/authentication/authn_jwt/identity_provider_factory.rb

```diff
@@ -5,20 +5,23 @@ module AuthnJwt
     # If the Identity variable is not configured and there is account field in url factory returns url provider
     # If the above conditions are not met exception is raised
     class IdentityProviderFactory
-      attr_accessor :from_url_provider, :from_decoded_token_provider
-
-      def initialize(authentication_parameters)
-        @from_url_provider = Authentication::AuthnJwt::IdentityFromUrlProvider.new(authentication_parameters)
-        @from_decoded_token_provider = Authentication::AuthnJwt::IdentityFromDecodedTokenProvider.new(authentication_parameters)
-      end
+      def self.create_identity_provider(
+        authentication_parameters:,
+        identity_from_url_provider_class: Authentication::AuthnJwt::IdentityFromUrlProvider,
+        identity_from_decoded_token_class: Authentication::AuthnJwt::IdentityFromDecodedTokenProvider
+      )
+        identity_provider = identity_from_decoded_token_class.new(authentication_parameters)
+        puts "decoded token identity avalible? = #{identity_provider.identity_available?}"
+        if identity_provider.identity_available?
+          Rails.logger.debug(LogMessages::Authentication::AuthnJwt::DECODED_TOKEN_IDENTITY_PROVIDER_SELECTED.new)
```

Comment:
`Rails.logger` should also be a dependency (i.e param with default value in method signature)